core/iwasm/common/wasm_application.c

```diff
@@ -657,23 +715,54 @@ wasm_application_execute_func(WASMModuleInstanceCommon *module_inst,
 #endif
                 break;
             }
-#endif
-#if WASM_ENABLE_SIMD != 0
-            case VALUE_TYPE_V128:
+#endif /* WASM_ENABLE_GC == 0 && WASM_ENABLE_REF_TYPES != 0 */
+            default:
             {
-                uint64 *v = (uint64 *)(argv1 + k);
-                os_printf("<0x%016" PRIx64 " 0x%016" PRIx64 ">:v128", *v,
-                          *(v + 1));
-                k += 4;
-                break;
+#if WASM_ENABLE_GC != 0
+                if (wasm_is_type_reftype(type->types[type->param_count + j])) {
```

Comment:
Should we extract this code piece (and also in new other code) to a function for better readability ?

Reply:
Yes, currently we just enable the patch to pass most spec cases, some codes may look a little ugly and need to be polished in the future.